internal/compiler/output_columns.go

```diff
@@ -97,18 +97,26 @@ func outputColumns(qc *QueryCatalog, node ast.Node) ([]*Column, error) {
 			}
 
 		case *pg.CoalesceExpr:
```

Comment:
This behavior is improved, but still a bit strange. We don't really know what the return type will be until typechecking lands, which is far off.